luigi/notifications.py

```diff
@@ -234,20 +231,21 @@ def send_email_ses(sender, subject, message, recipients, image_png):
 
 def send_email_sendgrid(sender, subject, message, recipients, image_png):
     import sendgrid as sendgrid_lib
-    client = sendgrid_lib.SendGridClient(
-        sendgrid().username, sendgrid().password, raise_errors=True)
-    to_send = sendgrid_lib.Mail()
-    to_send.add_to(recipients)
-    to_send.set_from(sender)
-    to_send.set_subject(subject)
+    client = sendgrid_lib.SendGridAPIClient(sendgrid().apikey)
     if email().format == 'html':
-        to_send.set_html(message)
+        to_send = sendgrid_lib.Mail(from_email=sender,
+            to_emails=recipients,
+            subject=subject,
+            html_content=message)
     else:
-        to_send.set_text(message)
+        to_send = sendgrid_lib.Mail(from_email=sender,
+            to_emails=recipients,
+            subject=subject,
+            plain_text_content=message)
```

Comment:
The `Mail()` generation could be more DRY (similar to how it was before):
```python
sendgrid_mail = sendgrid_lib.Mail()

sendgrid_mail.from_email = sender
sendgrid_mail.to = receipients
sendgrid_mail.subject = subject

if email().format == 'html':
    sendgrid_mail.add_content(message, 'text/html')
else:
    sendgrid_mail.add_content(message, 'text/plain')

if image_png:
    sendgrid_mail.add_attachment(image_png)
```

Reply:
how about initializing `Mail()` with the sender/recipients/subject already? would this be enough?

I originally had it that way with `add_content` dependent on html or not, but I wasn't sure `add_content` would accept `'text/html'` as the mime type (and have no way of testing this currently).

Reply:
sure, you can initialize mail with sender/receipients/subject before the conditional

As far as the `add_content`, i traced the source code and provided the same values they were are.

Reply:
ah, perfect. I guess I was one level too shallow. I'll fix this up and commit again. Is there a way to squash the commits after i've submitted the PR or do I have to start a new PR? (sorry, haven't had much experience with this)

Reply:
you can just push new commits. We tend to squash merge PRs